### PR TITLE
Added new methods num_timesteps, num_trajectories and __repr__ to pri…

### PR DESCRIPTION
**Pull request by knutfrode**
…nt summary information about a dataset (print self.ds.traj). New commandline utility trajaninfo <filename> also uses this new repr.

### CODE REVIEW

File: trajan/traj.py [hunk 1]
```diff
@@ -63,6 +64,44 @@ def __init__(self, ds, obsdim, timedim):
...
 
+    def num_timesteps(self):
+        if self.timedim in self.ds.sizes:
+            timedim = self.timedim
+        else:
+            logger.warning(f'self.timedim ({self.timedim}) is not an existing dimension! Using instead "obs".')
+            timedim = 'obs'
+        return self.ds.sizes[timedim]
+
+    def num_trajectories(self):
```
Comment (gauteh) [inline trajan/traj.py:75]: Is this one necessary? Why not use sizes directly?

Comment (knutfrode) [reply trajan/traj.py:75]: Yes, you are right. I will do that. However, we then have to check whether time or obs ?

And btw: I found for one dataset that self.timedim was "time", whereas the dataset only had "obs" as a dimension. Looks like an existing error?

**Comment by knutfrode** [comment]
Hm, I don't understand why these changes should lead to this test failing
https://github.com/OpenDrift/trajan/blob/main/tests/test_dims.py#L9

**Comment by gauteh** [comment]
> > Hm, I don't understand why these changes should lead to this test failing https://github.com/OpenDrift/trajan/blob/main/tests/test_dims.py#L9
> 
> Now I understand: in that test the dataset/trajectory was averaged over the time dimension, and afterwards the repr method was invoked with `print(b.traj)`. This special case is now handled, but I guess averaging trajectories over time is not a very common operation?

Doesn't that give you the mean position? I do that quite often, e.g. for a mooring or when I have a 20 minute wave measurement and need to plot the average position for that.

File: trajan/traj.py (hunk 1 above)
Comment (gauteh) [reply trajan/traj.py]: Was that the averaged dataset?

**Comment by knutfrode** [comment]
> > > Hm, I don't understand why these changes should lead to this test failing https://github.com/OpenDrift/trajan/blob/main/tests/test_dims.py#L9
> > 
> > 
> > Now I understand: in that test the dataset/trajectory was averaged over the time dimension, and afterwards the repr method was invoked with `print(b.traj)`. This special case is now handled, but I guess averaging trajectories over time is not a very common operation?
> 
> Doesn't that give you the mean position? I do that quite often, e.g. for a mooring or when I have a 20 minute wave measurement and need to plot the average position for that.

Yes, it makes sense for that purpose, but then I guess we do not anymore have a trajectory dataset?
But at least repr now also handles this special case, and reports that there is no time dimension present.

File: trajan/traj.py [hunk 2]
```diff
@@ -63,6 +64,39 @@ def __init__(self, ds, obsdim, timedim):
...
 
+    def __repr__(self):
+        output = '=======================\n'
+        output += 'TrajAn info:\n'
+        output += '------------\n'
+        output += f'{self.ds.sizes["trajectory"]} trajectories\n'
+        if self.timedim in self.ds.sizes:
+            output += f'{self.ds.sizes[self.timedim]} timesteps\n'
+        else:
+            logger.warning(f'self.timedim ({self.timedim}) is not an existing dimension!\n')
```
Comment (gauteh) [inline trajan/traj.py:75]: In this case the number of time steps is 1?

**Comment by gauteh** [comment]
> > > > Hm, I don't understand why these changes should lead to this test failing https://github.com/OpenDrift/trajan/blob/main/tests/test_dims.py#L9
> > > 
> > > 
> > > Now I understand: in that test the dataset/trajectory was averaged over the time dimension, and afterwards the repr method was invoked with `print(b.traj)`. This special case is now handled, but I guess averaging trajectories over time is not a very common operation?
> > 
> > 
> > Doesn't that give you the mean position? I do that quite often, e.g. for a mooring or when I have a 20 minute wave measurement and need to plot the average position for that.
> 
> Yes, it makes sense for that purpose, but then I guess we do not anymore have a trajectory dataset? But at least repr now also handles this special case, and reports that there is no time dimension present.

I think we do, just one with a single onservation (so far).

File: trajan/traj.py (hunk 2 above)
Comment (knutfrode) [reply trajan/traj.py:75]: Would you call it a timestep if there is no associated time?

Comment (gauteh) [reply trajan/traj.py:75]: Hehe good point, maybe observation. We still have one sample. It's a pity that time is lost when averaged.

File: trajan/traj.py (hunk 1 above)
Comment (knutfrode) [reply trajan/traj.py]: Yes, I guess it was. But this should be handled now - same problem as discussed below.

**Comment by knutfrode** [comment]
At least we have now a working version without failing tests, and the `repr` method can be improved/polished later.
Thus merging this now.

**Pull request merged**